### PR TITLE
feat: add GroupListPicker component

### PR DESCRIPTION
**Pull request by djamaile**
Signed-off-by: djamaile <rdjamaile@gmail.com>

## Hey, I just made a Pull Request!

new

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx
Comment (tudi2d) [inline plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx:140]: just pointing out because I'm already at it - this could be smth. like

```suggestion
          <PeopleIcon fontSize="large" style={(theme: BackstageTheme) => ({ marginRight: theme.spacing(1) }) } />
```

not to important - [here is the doc](https://v4.mui.com/customization/spacing/)

File: plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx
Comment (tudi2d) [inline plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx:30]: is it possible to use Theme variables for `fontSize`,`fontWeigth`,... ? :)

File: plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx
Comment (tudi2d) [reply plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx:78]: Ah - I see that you have this from the doc, right: https://v4.mui.com/components/popover/#popover? Still curious what would happen if you remove it :/

**Review by tudi2d — COMMENTED** [review]
like this - some smaller things - hopefully I'm not to picky here. In general - did you consider splitting up the component in e.g. a `GroupListPickerPopover` (containing only the Popover & relevant state) & `GroupListPicker`? :)

File: plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx
Comment (tudi2d) [inline plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx:140]: think you have to import it - sorry 😬

File: plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx
Comment (djamaile) [reply plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx]: You're right, made it optional!

File: plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx
Comment (djamaile) [reply plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx]: 🧹

File: plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx
Comment (djamaile) [reply plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx]: 🧹

File: plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx
Comment (djamaile) [reply plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx]: did it for `fontWeight` and used rem for `fontSize` instead.

**Comment by djamaile** [comment]
PRs can get stale?! Def not stale, just waiting on a review.

File: plugins/catalog-react/src/components/GroupListPicker/index.ts
Comment (Rugvip) [inline plugins/catalog-react/src/components/GroupListPicker/index.ts:18]: Is the button exported intentionally? It feels like it would be enough to export the picker itself

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thanks! 👍 Couple of nits

Another thing I'm wondering is how you feel about the placement of this in `catalog-react` vs something like `org-react`. It feels a bit off to have kind-specific components in `catalog-react` tbh, it's already a pretty large library. It also conflicts quite heavily with the `UserListPicker`, which is a different type of component that helps filter catalog index pages.

File: plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx
Comment (mathiasbronner) [reply plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx:103]: true, should be `300px`

**Comment by djamaile** [comment]
@Rugvip I agree that `catalog-react` might be an awkward place. Do you mean creating a new plugin called `org-react` or just the `org` plugin?

**Comment by Rugvip** [comment]
@djamaile yep a new `org-react` package. Same relation between that and `org` as between `catalog-react` and `catalog`. They're not separate plugins, but library packages provided by the plugins for others to use.

File: .github/CODEOWNERS
Comment (djamaile) [inline .github/CODEOWNERS:52]: @Rugvip Is this the right owner? Does it need an owner at all?

Comment (Rugvip) [reply .github/CODEOWNERS:52]: You can skip the explicit owner, there's a catch-all at the bottom that handles this

File: plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx
Comment (Rugvip) [reply plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx]: As someone using this component in my own page, how do I know what group the user ended up picking?

**Comment by djamaile** [comment]
@Rugvip its all green, feel free to review again 😸

File: plugins/org-react/src/plugin.ts
Comment (Rugvip) [inline plugins/org-react/src/plugin.ts:23]: This package isn't a plugin package and shouldn't export a plugin. It's a library, so you can export the component straight away without the extension wrapping.

File: plugins/org-react/src/components/GroupListPicker/GroupListPicker.tsx
Comment (Rugvip) [inline plugins/org-react/src/components/GroupListPicker/GroupListPicker.tsx:39]: This one is a bit weird to me, it seems like it just sets the label of the button and doesn't actually have an impact on which group is selected? Perhaps it should be dropped and have it be a completely uncontrolled component instead, with it displaying the currently selected team. It also conflicts a bit with the placeholder I think? even though that one's for the dropdown

**Review by Rugvip — CHANGES_REQUESTED** [review]
🎉 

CC @backstage/maintainers for the addition of the `org-react` package

File: plugins/org-react/src/plugin.ts
Comment (djamaile) [reply plugins/org-react/src/plugin.ts]: Updated!

File: plugins/org-react/src/routes.ts
Comment (Rugvip) [inline plugins/org-react/src/routes.ts:20]: Should remove this too

**Review by Rugvip — CHANGES_REQUESTED** [review]
Another round! 🍻

File: plugins/org-react/src/components/GroupListPicker/GroupListPicker.tsx
Comment (djamaile) [reply plugins/org-react/src/components/GroupListPicker/GroupListPicker.tsx:87]: So, I remember when I was working with Backstage (internally, prev company) I saw that on the user page that components were displayed with the namespace in the name. So, I thought that might be the standard. But honestly, thinking back I think that might've been just an internal setting we enabled? But for this specific case I think you are right, that is actually noise. I just went to check how it rendered and it's quite large. So, removing it for now! 😸

File: plugins/org-react/README.md
Comment (djamaile) [reply plugins/org-react/README.md]: 🧹

File: plugins/org-react/src/components/GroupListPicker/GroupListPicker.test.tsx
Comment (djamaile) [reply plugins/org-react/src/components/GroupListPicker/GroupListPicker.test.tsx]: 🧹

File: plugins/org-react/src/components/GroupListPicker/GroupListPickerButton.tsx
Comment (djamaile) [reply plugins/org-react/src/components/GroupListPicker/GroupListPickerButton.tsx]: Yeah missed that, refactored it to a `MUI` button 🧹

File: plugins/org-react/src/routes.ts
Comment (djamaile) [reply plugins/org-react/src/routes.ts]: 🧹

**Comment by mathiasbronner** [comment]
whats the status now btw @Rugvip @djamaile? Just asking because i'm looking forward to using it soon 🥳

**Comment by djamaile** [comment]
@mathiasbronner Sorry, for the long wait! Besides me making silly mistakes we also had our offsite + BackstageCon in Detroit. So, things have been slower than normal. I hope to get this in the next release 😄

**Review by Rugvip — APPROVED** [review]
Let's :shipit: 😁

**Comment by suuus** [comment]
Thank you for contributing during Hacktoberfest 🍂! You can claim the Backstage Hacktoberfest Holopin at https://bck.st/hacktoberfest-holopin 🙌🏻

**Pull request merged**
